### ISSUE TITLE
Cypress adds __ to redirect URL after AJAX call

### ISSUE THREAD

**Issue by intoeetive** [cypress-io/cypress#6749]
### Current behavior:

I have a page that makes an AJAX call and then performs javascript redirect by setting `window.location` to URL which comes as javascript variable (ex. `test2.html`. That URL is relative, without leading slash. On the tests, URL is getting `__/` prepended. That is, test page redirect to `http://localhost/__/test2.html`

### Desired behavior:

Page should redirect to `http://localhost/test2.html`

### Test code to reproduce

```js
context('Test underscore', () => {
it('__', () => {
		cy.wait(1000)
		cy.visit('test.html')
		cy.wait(1000)
		cy.contains('some text')
   })
})
```

```html
 <html>
 <head>
 <script type="text/javascript" src="https://code.jquery.com/jquery-3.4.1.min.js"></script>
 <script type="text/javascript">
 var sync_baseurl = 'test2.html';
 	$.ajax({
		url: '/',
 		type: 'POST',
 		dataType: 'json',
 		data: {
 			"test": "test"
 		},
 		beforeSend: function(xhr, settings) {
 			// Increment the running count
 		},
 		complete: function(xhr, textStatus) {
 			// Decrement the running count
 			finish_sync();
 		},
 		success: function(data, textStatus, xhr) {
 
 		}
 	});
 
 	finish_sync = function() {
 			window.location = sync_baseurl;
 	};
 
 </script>
 </head>
 <body>
 init
 </body>
</html>
```

### Versions

Cypress 4.1.0
Chrome 80
Windows 10

**Comment by jennifer-shehane** [comment]
Duplicate of #3975

**Comment by jennifer-shehane** [comment]
This fix is available starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments) which you can access by setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

The fix is experimental, so there may be some situations where the this is not fixed. 

If you're still this issue while setting the `experimentalSourceRewriting` to true in 4.6.0 - [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a reproducible example + screenshots, etc - filling out our issue template.
